components/patient-data/ui/src/main/resources/PhenoTips/PatientConsent.xml

```diff
@@ -244,17 +177,461 @@ $services.localization.render('PhenoTips.PatientConsent_message', ["$!{formUrl}"
+              this.classVariables.successfulLoadCallback();
+            }
+            var continueButton = $('#patient-consent-continue');
+            continueButton.removeClass("hidden");
+            this.classVariables.consentsBlock.initialized = true;
+          } else {
+            this.classVariables.errorFunction();
+          }
+          this.classVariables.consentsBlock.show();
+        },
+
+        displayConsents: function (consentsBlock, action, consents)
+        {
+          var listBlock = consentsBlock.children(".list-block");
+          if (listBlock.length == 0) {
+            listBlock = $("&lt;div class='list-block consents'&gt;");
+            consentsBlock.append(listBlock);
+          }
+          var list;
+          if (action == "edit") {
+            var _this = this;
+            list = $.map(consents, function (consent)
+            {
+              return _this.displayGrantableConsent(consent);
+            })
+          } else if (action == "view") {
+            var granted = this.filterGranted(consents);
+            this.displayViewTitle(consentsBlock, granted);
+            list = granted.map(this.displayConsentDescription);
+          }
+          if (list) {
+            var listHtml = $("&lt;ul&gt;&lt;/ul&gt;");
+            list.each(function (c)
+            {
+              listHtml.append(c)
+            });
+            listBlock.append(listHtml);
+          }
+        },
+
+        /** @return the list of ids, whose corresponding checkboxes are checked */
+        listUIGranted: function(consentsBlock)
+        {
+          var _this = this;
+          var block = consentsBlock;
+          if (!block) {
+            block = this.classVariables.consentsBlock;
+          }
+          var checked = block.find("input[type=checkbox]").filter(function(i, ch){
+            return $(ch).is(":checked")
+          });
+          return checked.map(function(i, ch) {
+            var id = $(ch).attr("id");
+            return id.replace(_this.consentIdSuffix, "");
+          }).get();
+        },
+
+        attachManagingListeners: function (baseUrl)
+        {
+          var _this = this;
+          var listBlock = $(".list-block");
+          listBlock.on("change", "li input", function (event)
+          {
+            var checkbox = $(event.target);
+            var rest = _this.interactWithManagingRest(checkbox.attr("data-bind"), baseUrl);
+            if (checkbox.is(":checked")) {
+              rest("grant");
+            } else {
+              rest("revoke");
+            }
+          })
+        },
+
+        interactWithManagingRest: function (consentId, baseUrl)
+        {
+          var _this = this;
+          return function (action)
+          {
+            var url = baseUrl + action;
+            $.ajax(url, {data: consentId, dataType: "text", method: "PUT", contentType: "text/plain;charset=UTF-8"})
+              .fail(_this.classVariables.errorFunction);
```

Comment:
Right, I didn't write that. No sir. Wasn't me. :)

That's Andrey's code into which I was adding, hence the Prototype.